ui/src/components/policy/PolicyList.js

```diff
@@ -66,15 +98,38 @@ export default class PolicyList extends React.Component {
         this.api = props.api;
         this.onSubmitDeletePolicy = this.onSubmitDeletePolicy.bind(this);
         this.onCancelDeletePolicy = this.onCancelDeletePolicy.bind(this);
+        this.onSubmitDuplicatePolicy = this.onSubmitDuplicatePolicy.bind(this);
+        this.onCancelDuplicatePolicy = this.onCancelDuplicatePolicy.bind(this);
         this.toggleAddPolicy = this.toggleAddPolicy.bind(this);
         this.reloadPolicies = this.reloadPolicies.bind(this);
+        this.closeModals = this.closeModals.bind(this);
         this.closeModal = this.closeModal.bind(this);
+        this.policyListToMap = this.policyListToMap.bind(this);
+
         this.state = {
-            list: props.policies || [],
+            policiesMap: this.policyListToMap(props.policies),
             showAddPolicy: false,
+            showDelete: false,
+            deletePolicyName: null,
+            showDuplicatePolicy: false,
+            duplicatePolicyName: null,
+            duplicateVersionSourceName: null,
+            duplicateVersionName: null,
         };
     }
 
+    policyListToMap(policies) {
+        let policyList = policies || [];
+        return policyList.reduce(function (map, obj) {
+            if (!map[obj.name]) {
+                map[obj.name] = [];
+            }
+
+            map[obj.name].push(obj);
+            return map;
+        }, {});
+    }
+
     onSubmitDeletePolicy() {
```

Comment:
When versions are not expanded and clicking on delete - all policy versions will be deleted.